src/cmd-buildprep

```diff
@@ -143,19 +150,25 @@ class S3Fetcher(Fetcher):
 
     def __init__(self, url_base):
         super().__init__(url_base)
+        self.s3 = boto3.client('s3')
+        self.s3config = boto3.s3.transfer.TransferConfig(
+            num_download_attempts=5
+        )
 
+    @retry(stop=retry_stop, retry=retry_s3_exception)
     def fetch_impl(self, url, dest):
-        subprocess.check_call(['aws', 's3', 'cp', url, dest],
-                              stdout=subprocess.DEVNULL)
+        assert url.startswith("s3://")
+        bucket, key = url[len("s3://"):].split('/', 1)
+        self.s3.download_file(bucket, key, dest, Config=self.s3config)
```

Comment:
I'm not sure if these would be the same errors, there is no proper documentation on what are these download attempts in the config.

I'd rather keep several attempts just in case

Reply:
Hmm, might be worth digging into that. Unless it's a completely different set of transient errors, I don't think we should retry 25 times for the same errors.

Reply:
Did you get a chance to look into this one?

Reply:
Looking at it now - I'm not sure if my testing has caused the build to retry yet, moved this to TODO in PR description

Reply:
Fixed this - `download_file` does not need to be retried with decorator (it traps the same exceptions), but other calls should be

Reply:
Cool, thanks for investigating. Maybe let's add a comment about this near `num_download_attempts` ?

Reply:
Done